### ISSUE TITLE
Seg fault on some inputs

### ISSUE THREAD

**Issue by quidryan** [openllb/hlb#9]
This produces a seg fault: 

```
fs default() {
  scratch
  nothing fs { scratch; }
}
fs nothing(fs repo) {
    scratch
}
```

**Comment by quidryan** [comment]
Sample output:
```
➜  managed-artifacts-generator-baseos git:(master) ✗ hlb run syntaxerror.hlb
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x478b48c]
goroutine 1 [running]:
github.com/openllb/hlb/codegen.emitChainStmt.func1(0x4982ec0, 0xc000325d40, 0x4993ecf, 0x2)
        /go/src/hlb/codegen/codegen.go:165 +0xdc
github.com/openllb/hlb/codegen.emitBlock(0xc0003ef250, 0xc00033ca80, 0x4993ecf, 0x2, 0xc0003ef2d0, 0x2, 0x2, 0x49cdaf8, 0x8, 0xc0004f5578, ...)
        /go/src/hlb/codegen/codegen.go:146 +0x529
github.com/openllb/hlb/codegen.emitFilesystemBlock(0xc0003ef250, 0xc00033ca80, 0xc0003ef2d0, 0x2, 0x2, 0x49cdaf8, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/hlb/codegen/codegen.go:185 +0xbd
github.com/openllb/hlb/codegen.emitFuncDecl(0xc0003ef250, 0xc00033ca20, 0xc000372360, 0xc000318700, 0x0, 0x0, 0x49cdaf8, 0xc00041ccd0, 0xc0004f5808, 0x40df6fb, ...)
        /go/src/hlb/codegen/decl.go:44 +0x717
github.com/openllb/hlb/codegen.emitFilesystemFuncDecl(0xc0003ef250, 0xc00033ca20, 0xc000372360, 0xc000318700, 0x49cdaf8, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/hlb/codegen/decl.go:55 +0xa0
github.com/openllb/hlb/codegen.Generate(0xc000318700, 0xc00033ca00, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /go/src/hlb/codegen/codegen.go:52 +0x4c0
github.com/openllb/hlb.Compile(0x4ac5160, 0xc0000de000, 0xc0000d6050, 0x499727c, 0x7, 0xc0004f5ca8, 0x1, 0x1, 0x0, 0x0, ...)
        /go/src/hlb/hlb.go:41 +0x242
github.com/openllb/hlb/cmd/hlb/command.glob..func4(0xc0003bb680, 0x9, 0x10)
        /go/src/hlb/cmd/hlb/command/run.go:86 +0x22d
github.com/urfave/cli/v2.(*Command).Run(0x50fe4c0, 0xc0003bb5c0, 0x0, 0x0)
        pkg/mod/github.com/urfave/cli/v2@v2.1.1/command.go:161 +0x499
github.com/urfave/cli/v2.(*App).RunContext(0xc0000fe300, 0x4ac5160, 0xc0000de000, 0xc0000da000, 0x3, 0x3, 0x0, 0x0)
        pkg/mod/github.com/urfave/cli/v2@v2.1.1/app.go:302 +0x791
github.com/urfave/cli/v2.(*App).Run(...)
        pkg/mod/github.com/urfave/cli/v2@v2.1.1/app.go:211
main.main()
        /go/src/hlb/cmd/hlb/main.go:12 +0x69
```

**Comment by quidryan** [comment]
Super low priority, just pointing it out.

**Comment by coryb** [comment]
resolved via #27
